templates/web.template.yml

```diff
@@ -182,7 +181,7 @@ run:
       hook: yarn
       cmd:
         - |-
-          if [ "$version" != "tests-passed" ]; then
+          if [ "$version" != "tests-passed" ] && [ "$clean_up_node_modules" == true ]; then
```

Comment:
I'll push a commit with updated logic

Reply:
Does this mean nothing is broken if stable had been working with this logic?

Reply:
We originally added this to workaround an edge-case with a single dependency in yarn. I guess that edge-case no longer exists, which is why this hasn't been causing a problem.

Now, we have a new problem with the change from yarn -> pnpm. This one is not an edge-case - it's 100% guaranteed to blow up. But, it hasn't affected anyone yet, because the current launcher base image is still yarn-based.

Here's a PR with improved logic: https://github.com/discourse/discourse_docker/pull/869